BLE Indoor Positioning/src/test/java/com/nexenio/bleindoorpositioning/ble/advertising/AdvertisingPacketUtilTest.java

```diff
@@ -0,0 +1,319 @@
+        long endTimestamp = latestTimestamp - 1;
+
+        List<IBeaconAdvertisingPacket> packagesBetween = AdvertisingPacketUtil.getAdvertisingPacketsBetween(advertisingPackets, startTimestamp, endTimestamp);
+
+        for (AdvertisingPacket advertisingPacket : packagesBetween) {
+            assertTrue("Timestamp is smaller than our start timestamp", advertisingPacket.timestamp >= startTimestamp);
+            assertTrue("Timestamp is greater than end timestamp", advertisingPacket.timestamp <= endTimestamp);
+        }
+
+        assertEquals("Incorrect number of packages returned", 1, packagesBetween.size());
+    }
+
+    @Test
+    public void getAdvertisingPacketsBetween_endIndexIsCalculatedInSecondHalf_returnsCorrectList() {
+        long latestTimestamp = System.currentTimeMillis();
+        int numberOfPackets = 3;
+        advertisingPackets = getAdvertisingPackets(1, numberOfPackets, latestTimestamp);
+
+        IBeaconAdvertisingPacket secondToLastPacket = advertisingPackets.get(numberOfPackets - 2);
+
+        long startTimestamp = secondToLastPacket.timestamp;
+        long endTimestamp = latestTimestamp - 1;
+
+        List<IBeaconAdvertisingPacket> packagesBetween = AdvertisingPacketUtil.getAdvertisingPacketsBetween(advertisingPackets, startTimestamp, endTimestamp);
+
+        for (AdvertisingPacket advertisingPacket : packagesBetween) {
+            assertTrue("Timestamp is smaller than our start timestamp", advertisingPacket.timestamp >= startTimestamp);
+            assertTrue("Timestamp is greater than end timestamp", advertisingPacket.timestamp <= endTimestamp);
+        }
+
+        assertEquals("Incorrect number of packages returned", 1, packagesBetween.size());
+    }
+
+    public List<IBeaconAdvertisingPacket> getAdvertisingPackets() {
+        return getAdvertisingPackets(10, 1000, System.currentTimeMillis());
+    }
+
+    /**
+     * @param packetsFrequency Frequency in which packets timestamp should be created (in Hertz)
+     * @param packetsCount     Number of packets to return
+     * @return List of advertising packets
+     */
+    public ArrayList<IBeaconAdvertisingPacket> getAdvertisingPackets(int packetsFrequency, int packetsCount, long latestPacketTimestamp) {
+        ArrayList<IBeaconAdvertisingPacket> advertisingPackets = new ArrayList<>();
+
+        int timestampDelta = (int) TimeUnit.SECONDS.toMillis(1) / packetsFrequency;
+        long oldestPacketTimestamp = latestPacketTimestamp - ((packetsCount - 1) * timestampDelta);
+
+        for (int packetIndex = 0; packetIndex < packetsCount; packetIndex++) {
+            IBeaconAdvertisingPacket advertisingPacket = new IBeaconAdvertisingPacket(BeaconTest.IBEACON_ADVERTISING_DATA);
+            advertisingPacket.setTimestamp(oldestPacketTimestamp + (packetIndex * timestampDelta));
+            advertisingPackets.add(advertisingPacket);
+        }
+
+        return advertisingPackets;
+    }
+
+    // Benchmarking
```

Comment:
This seems to be a lot of stuff that is not needed in the regular unit tests. Why do we move this to a new test class?